File: app/src/main/java/com/example/milkaggregatorapplication/RetrofitInstance.java
package com.example.milkaggregatorapplication;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {

    private static String BASE_URL="https://graph.facebook.com/";
    private static String VERSION="v15.0/";

    private static Retrofit retrofit;

    public static Retrofit getRetrofitClient(){

        if(retrofit==null)
        {
            retrofit=new Retrofit.Builder()
                    .baseUrl(BASE_URL+VERSION)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(new OkHttpClient().newBuilder().addInterceptor(new MyIntercepter()).build())
                    .build();

        }

        return retrofit;


    }
}

File: app/src/main/java/com/example/milkaggregatorapplication/RetrofitInstance.kt
package com.example.milkaggregatorapplication

import okhttp3.OkHttpClient.newBuilder
import okhttp3.OkHttpClient.Builder.addInterceptor
import okhttp3.OkHttpClient.Builder.build
import retrofit2.Retrofit
import com.example.milkaggregatorapplication.RetrofitInstance
import retrofit2.converter.gson.GsonConverterFactory
import okhttp3.OkHttpClient
import com.example.milkaggregatorapplication.MyIntercepter

object RetrofitInstance {
    private const val BASE_URL = "https://graph.facebook.com/"
    private const val VERSION = "v15.0/"
    private var retrofit: Retrofit? = null
    val retrofitClient: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL + VERSION)
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(OkHttpClient().newBuilder().addInterceptor(MyIntercepter()).build())
                    .build()
            }
            return retrofit
        }
}